module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/HoldsTest.php

```diff
@@ -98,4 +101,322 @@ public function testGetSuppressedLocations(array $configArray, array $expectedLi
+                $availabilityStatus[$i][0],
+                $availabilityStatus[$i][1]
+            );
+            $i++;
+        }
+        $holdConfig = [
+            'function' => 'placeHold',
+            'HMACKeys' => ['id', 'location'],
+        ];
+
+        // Test 'all' mode (all items get a link)
+        $holdings = $method->invoke($logic, $result, 'all', $holdConfig);
+        $this->assertArrayHasKey('link', $holdings['holdings_id_1|Main Library'][0]);
+        $this->assertArrayHasKey('link', $holdings['holdings_id_2|Secondary Library'][0]);
+
+        // Test 'holds' mode (only available items get a link)
+        $holdings = $method->invoke($logic, $result, 'holds', $holdConfig);
+        $this->assertArrayNotHasKey('link', $holdings['holdings_id_1|Main Library'][0]);
+        $this->assertArrayHasKey('link', $holdings['holdings_id_2|Secondary Library'][0]);
+
+        // // Test 'recalls' mode (only unavailable items get a link)
+        $holdings = $method->invoke($logic, $result, 'recalls', $holdConfig);
+        $this->assertArrayHasKey('link', $holdings['holdings_id_1|Main Library'][0]);
+        $this->assertArrayNotHasKey('link', $holdings['holdings_id_2|Secondary Library'][0]);
+    }
+
+    /**
+     * Test getRequestDetails method
+     *
+     * @return void
+     */
+    public function testGetRequestDetails(): void
+    {
+        $hmac = $this->createMock(HMAC::class);
+        $hmac->method('generate')->willReturn('test-hash-key');
+
+        $logic = $this->getHoldsLogic(hmac: $hmac);
+
+        $reflection = new \ReflectionClass($logic);
+        $method = $reflection->getMethod('getRequestDetails');
+        $method->setAccessible(true);
+
+        $details = [
+            'id' => 'test123',
+            'location' => 'Main Library',
+            'source' => 'Solr',
+        ];
+
+        $hmacKeys = ['id', 'location'];
+        $action = 'Hold';
+
+        $result = $method->invoke($logic, $details, $hmacKeys, $action);
+
+        $this->assertEquals('Hold', $result['action']);
+        $this->assertEquals('test123', $result['record']);
+        $this->assertEquals('Solr', $result['source']);
+        $this->assertStringContainsString('hashKey=test-hash-key', $result['query']);
+        $this->assertEquals('#tabnav', $result['anchor']);
+
+        // Test with link overrides
+        $linkOverrides = ['id' => 'override123', 'source' => 'Override'];
+        $result = $method->invoke($logic, $details, $hmacKeys, $action, $linkOverrides);
+
+        $this->assertEquals('override123', $result['record']);
+        $this->assertEquals('Override', $result['source']);
+    }
+
+    /**
+     * Test getHoldingsGroupKey method
+     *
+     * @return void
+     */
+    public function testGetHoldingsGroupKey(): void
+    {
+        // Test default grouping
+        $logic = $this->getHoldsLogic();
+
+        $reflection = new \ReflectionClass($logic);
```

Comment:
I see there are several more direct uses of ReflectionClass here that might be able to be simplified with the ReflectionTrait. If the trait has any limitations, let me know -- might be worth improving it as needed to keep the logic separate.

Reply:
This too should be addressed with recent commit using ReflectionTrait